src/wp-includes/class-wp-scripts.php

```diff
@@ -294,15 +294,42 @@ public function do_item( $handle, $group = false ) {
 			$cond_after  = "<![endif]-->\n";
 		}
 
+		$strategy = $this->get_eligible_loading_strategy( $handle );
+
 		$before_handle = $this->print_inline_script( $handle, 'before', false );
-		$after_handle  = $this->print_inline_script( $handle, 'after', false );
 
 		if ( $before_handle ) {
 			$before_handle = sprintf( "<script%s id='%s-js-before'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), $before_handle );
 		}
 
-		if ( $after_handle ) {
-			$after_handle = sprintf( "<script%s id='%s-js-after'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), $after_handle );
+		$after_handle = '';
+		if ( '' === $strategy ) {
+			$after_handle = $this->print_inline_script( $handle, 'after', false );
+
+			if ( $after_handle ) {
+				$after_handle = sprintf( "<script%s id='%s-js-after'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), $after_handle );
+			}
+		} else {
+			$after_standalone_handle = $this->print_inline_script( $handle, 'after-standalone', false );
+
+			if ( $after_standalone_handle ) {
+				$after_handle .= sprintf( "<script%s id='%s-js-after'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), $after_standalone_handle );
+			}
+
+			$after_non_standalone_handle = $this->print_inline_script( $handle, 'after-non-standalone', false );
+
+			if ( $after_non_standalone_handle ) {
+				$initial_type_attr = $this->type_attr;
+				$this->type_attr   = " type='text/template'";
+				$after_handle     .= sprintf(
+					'<script%1$s id=\'%2$s-js-after\' data-wp-executes-after=\'%2$s\'>%4$s%3$s%4$s</script>%4$s',
+					$this->type_attr,
+					esc_attr( $handle ),
+					$after_non_standalone_handle,
+					PHP_EOL
+				);
+				$this->type_attr   = $initial_type_attr;
```

Comment:
1. We probably don't need to modify the class property just to reset it as soon as we're done here.
2. In other places, we're using `\n` for new lines rather than `PHP_EOL`. I don't see many uses of `PHP_EOL` in core, so let's swap these out.

🔢 Apply same suggestion to `print_inline_script` below.

```suggestion
				$after_handle     .= sprintf(
					"<script%1$s id=\'%2$s-js-after\' data-wp-executes-after=\'%2$s\'>\n%3$s\n</script>\n",
					" type='text/template'",
					esc_attr( $handle ),
					$after_non_standalone_handle,
				);
```

Reply:
@joemcgill  Same comment as above ( https://github.com/10up/wordpress-develop/pull/54/files#r1168320548 )

Reply:
For the `PHP_EOL` issue you described above, I would suggest replacing your uses of `%4$s` with `\n` directly, rather than using a positional value.

Either way, can we address my first point about not resetting the `$this->type_attr` property unnecessarily?

Reply:
Resolved.